pyccel/ast/core.py

```diff
@@ -1349,26 +1349,30 @@ def __init__(self, iterable):
         self._iterable = iterable
         self._indices  = None
 
-        if isinstance(iterable, self.indexless_types):
-            self._num_indices_required =  0
+        if isinstance(iterable, PythonRange):
+            self._num_indices_required = 0
+        elif isinstance(iterable, PythonEnumerate):
+            self._num_indices_required = int(iterable.start != 0)
         elif isinstance(iterable, Product):
             self._num_indices_required = len(iterable.elements)
         elif isinstance(iterable, self.acceptable_iterator_types):
             self._num_indices_required = 1
         else:
-            raise TypeError("Unknown iterator type")
+            raise TypeError("Unknown iterator type {}".format(type(iterable)))
 
         super().__init__()
 
     @property
-    def num_indices_required(self):
-        """ Number of indices which must be generated in order to
+    def num_generated_iterators_required(self):
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/Iterator):

> A loop counter is sometimes also referred to as a loop iterator. A loop counter, however, only provides the traversal functionality and not the element access functionality.

I tend to agree with this distinction. In our case we are creating loop counters, I believe, rather than iterators. So I would call this property `num_loop_counters_required` or something similar. What do you think?

Reply:
That sounds right to me. Would you also like me to change `set_range_iterator` to `set_loop_counter`?

Reply:
Yes please

Reply:
Fixed 3b636b5